utils/serde/src/main/java/io/apicurio/registry/utils/serde/JsonSchemaKafkaDeserializer.java

```diff
@@ -0,0 +1,217 @@
+import io.apicurio.registry.rest.beans.ArtifactMetaData;
+import io.apicurio.registry.rest.beans.VersionMetaData;
+import io.apicurio.registry.utils.serde.util.Utils;
+
+/**
+ * @author eric.wittmann@gmail.com
+ */
+public class JsonSchemaKafkaDeserializer<T> extends AbstractKafkaSerDe<JsonSchemaKafkaDeserializer<T>> implements Deserializer<T> {
+
+    public static final String REGISTRY_JSON_SCHEMA_DESERIALIZER_VALIDATION_ENABLED = "apicurio.registry.serdes.json-schema.validation-enabled";
+
+    private static MedeiaJacksonApi api = new MedeiaJacksonApi();
+    private static ObjectMapper mapper = new ObjectMapper();
+    
+    private boolean validationEnabled = false;
+    private SchemaCache<SchemaValidator> schemaCache;
+
+    /**
+     * Constructor.
+     */
+    public JsonSchemaKafkaDeserializer() {
+        this(null);
+    }
+
+    /**
+     * Constructor.
+     * @param client
+     */
+    public JsonSchemaKafkaDeserializer(RegistryService client) {
+        super(client);
+    }
+
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#configure(java.util.Map, boolean)
+     */
+    @Override
+    public void configure(Map<String, ?> configs, boolean isKey) {
+        super.configure(configs);
+
+        Object ve = configs.get(REGISTRY_JSON_SCHEMA_DESERIALIZER_VALIDATION_ENABLED);
+        this.validationEnabled = Utils.isTrue(ve);
+
+        this.schemaCache = new SchemaCache<SchemaValidator>(getClient()) {
+            @Override
+            protected SchemaValidator toSchema(Response response) {
+                String schema = response.readEntity(String.class);
+                return api.loadSchema(new StringSchemaSource(schema));
+            }
+        };
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#deserialize(java.lang.String, byte[])
+     */
+    @Override
+    public T deserialize(String topic, byte[] data) {
+        throw new UnsupportedOperationException();
+    }
+    
+    /**
+     * @see org.apache.kafka.common.serialization.Deserializer#deserialize(java.lang.String, org.apache.kafka.common.header.Headers, byte[])
+     */
+    @Override
+    public T deserialize(String topic, Headers headers, byte[] data) {
+        if (data == null) {
+            return null;
+        }
+        
+        try {
+            JsonParser parser = mapper.getFactory().createParser(data);
+            if (validationEnabled) {
+                Long globalId = getGlobalId(headers);
+                
+                // If no globalId is provided, check the alternative - which is to check for artifactId and 
+                // (optionally) version.  If these are found, then convert that info to globalId.
+                if (globalId == null) {
+                    String artifactId = getArtifactId(headers);
+                    Integer version = getVersion(headers);
+                    if (version == null) {
+                        version = getLatestVersion(artifactId);
```

Comment:
No need for this call ... just diff between when looking up metadata when you have the version and when not == single call.